polar-core/src/bindings.rs

```diff
@@ -425,52 +427,91 @@ impl BindingManager {
     }
 }
 
+fn combine(left: &Symbol, l: Operation, right: &Symbol, r: Operation) -> Operation {
+    let unis = l
+        .constraints()
+        .into_iter()
+        .chain(r.constraints().into_iter())
+        .filter_map(|op| match op {
+            Operation {
+                operator: Operator::Unify,
+                args,
+            } if args.len() == 2
+                && args[0].value().as_symbol().is_ok()
+                && args[1].value().as_symbol().is_ok() =>
+            {
+                Some((
+                    args[0].value().as_symbol().unwrap().clone(),
+                    args[1].value().as_symbol().unwrap().clone(),
+                ))
+            }
+            _ => None,
+        })
+        .chain(std::iter::once((left.clone(), right.clone())));
+    let sub = crate::data_filtering::partition_equivs(unis)
+        .into_iter()
+        .flat_map(|cls| {
+            if let Some(rep) = cls.iter().find(|v| !v.is_temporary_var()) {
```

Comment:
I'm having a little trouble following what this is doing.

Seems like outputting _x => _x if there is no non-temporary variable in the equivalence set or _t => x if there is a non-temporary?